src/Form/molecules/FilterDropdown.tsx

```diff
@@ -0,0 +1,537 @@
+  const openLeft = (buttonRect.left + MIN_WIDTH) > window.innerWidth;
+  const openTop = (buttonRect.bottom + MIN_HEIGHT) > window.innerHeight;
+  const spaceTop = buttonRect.bottom > MIN_HEIGHT;
+
+  if (openLeft && openTop && spaceTop) {
+    position = 'top-left';
+  }
+
+  if (openTop && !openLeft && spaceTop) {
+    position = 'top-right';
+  }
+
+  if (!openTop && openLeft) {
+    position = 'bottom-left';
+  }
+
+  return position;
+}
+
+
+const isValueSelected = (item: IFilterItem, selected: IFilterDropdownValue) => {
+  let isItemSelected = false;
+
+  if (Array.isArray(selected)) {
+    selected.forEach((element: IFilterItem) => {
+
+      if (isListItem(element) && isListItem(item) && element.value === item.value) {
+        isItemSelected = true;
+      } else if (element === item) {
+        isItemSelected = true;
+      }
+    })
+
+  } else {
+
+    if (isListItem(selected) && isListItem(item)) {
+      isItemSelected = item.value === selected.value
+    } else {
+      isItemSelected = item === selected;
+    }
+
+  }
+
+  return isItemSelected;
+}
+
+// type checking
+// https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript
+
+const isListItem = (item: any): item is ListItem => {
+  if (item === null) { return false };
+
+  return (item.value !== undefined) && (item.text !== undefined);
+}
+
+const getNewSelected = (item: IFilterItem, selected: IFilterDropdownValue, optionType: IInputOptionsType): IFilterDropdownValue => {
+  let isItemSelected = false;
+
+  if (optionType === 'checkbox') {
+    const validSelected = Array.isArray(selected) ? selected : [];
+    if (typeof item === 'number') {
+      const newSelected: number[] = [];
+      validSelected.forEach((element: IFilterItem) => {
+        if (element === item) {
+          isItemSelected = true;
+        } else if (typeof element === 'number') {
+          newSelected.push(element);
+        }
+      })
+      if (!isItemSelected) {
+        newSelected.push(item);
+      }
+      return newSelected
+    }
+
+    if (typeof item === 'string') {
+      const newSelected: string[] = [];
+      validSelected.forEach((element: IFilterItem) => {
+        if (element === item) {
+          isItemSelected = true;
+        } else if (typeof element === 'string') {
+          newSelected.push(element);
+        }
+      })
+      if (!isItemSelected) {
+        newSelected.push(item);
+      }
+      return newSelected
+    }
+
+    const newSelected: ListItem[] = [];
+    validSelected.forEach((element: IFilterItem) => {
+      if (isListItem(element)) {
+        if (item.value === element.value) {
+          isItemSelected = true;
+        } else {
+          newSelected.push(element);
+        }
+      }
+    })
+    if (!isItemSelected) {
+      newSelected.push(item);
+    }
+    return newSelected
+  }
+
+  return item;
+}
+
+const getVisibleList = (list: IFilterItem[], maxItems: number, selected: IFilterDropdownValue): IFilterItem[] => {
```

Comment:
It would be good to see some basic comments to more easily identify what's going on with this function.